getndazn/dazn-lambda-powertools
# feat(middleware-correlation-ids): support sns sqs without raw deliveries

## What did you implement:

Closes #23 

Handle correlation IDs which are nested via SNS => SQS => Lambda integration in the middleware correlation ids package.
When raw message delivery is disabled in the SNS subscription, messages are delivered in JSON format along with SNS metadata along with the body of the message itself.

## How did you implement it:

If message attributes are not found in the SQS message payload, try to look for an SNS message payload and check if message attributes are set there.

## How can we verify it:

Create an SNS subscription with raw message delivery disabled with an SQS integration type which is has a lambda trigger. The original SNS message should have an correlation ID set in the message attributes which can then be picked up by the middleware now.

## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
